Dereg fixes

gamecoder95 opened the pull request:
- Made item num a string value
- If client is registered successfully, unregistered error messages don't appear
- Replaced "while True" with "while self._closed" for UDP and TCP servers and clients
- Added KeyboardInterrupt handling for server (note: I wasn't really able to test it as when it is running I can't seem to interrupt it in the python console anyway).

samratdebroy reviewed (approved):
looks good

42B commented on auctionsystem/server.py:
```diff
@@ -69,8 +70,12 @@ def __init__(self, recover=False):
+            self.loop.run_forever()
+            self.loop.close()
+        except KeyboardInterrupt:
+            self.loop.close()
```
Couldn't `self.loop.close()` be inside of `finally`?

```python
try:
    self.loop.run_forever()
except KeyboardInterrupt:
    sys.exit()
finally:
    self.loop.close()
```
?

42B commented on auctionsystem/tcp/client.py:
```diff
@@ -62,13 +64,17 @@ def get_tcp_client_socket():
+                handle_receive_cb(data, self.server_address)
+            except OSError:
+                await asyncio.sleep(0.1)
+                continue
```
If there isn't anything after this `except` clause, couldn't `continue` be removed?

samratdebroy reviewed (approved):
all good

gamecoder95 committed:
Set the item number to string

In bidding items

gamecoder95 committed:
Added field to prevent unregistered message bug

When registered, none of the unregistered or timeout messages appear on
the GUI.

gamecoder95 committed:
Fixed all the disconnection issues

Added the "closed" flag to everything and replaced the while loops with
checks for that closed bool.

gamecoder95 committed:
Added handling for keyboard interrupt for server

gamecoder95 committed:
Removed task.cancel() statements

Ain't nobody needing dat no mo.
